### PR TITLE
Add manifest var and release script

### PR DESCRIPTION
**Pull request by maleck13**
## Additional Information
The release in 1.3 will only need to tag the installation repo

## Verification Steps
Run the installation
Ensure the version is populated in the manifest in the cluster
``` oc get secret manifest -n webapp --template '{{index .data "generated_manifest"}}'  | base64 -d ```

Test the release script

- start on this branch and run the release script. Ensure to set your own origin
``` 
export REMOTE=mine
./scripts/release.sh -b v1.3 -r release-1.3.25-rc1-test

```

New v1.3 branch should exist on the remote and tag created
Run Again with new rc
```
export REMOTE=mine
./scripts/release.sh -b v1.3 -r release-1.3.25-rc2-test

```

### CODE REVIEW

**Comment by maleck13** [comment]
@mikenairn @david-martin  would be good to get a review if you get chance

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+#check the current branch we are on
+currentBranch=$(git branch | grep \* | cut -d ' ' -f2)
```
Comment (mikenairn) [inline scripts/release.sh]: You could also use `git symbolic-ref --short HEAD` or `git rev-parse --abbrev-ref HEAD`

File: scripts/release.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+if [[ "$carryON" == "y" ]]
+then
+echo "
+what branch do you want?"
+read  -p "Input: " branch
+echo "
+creating new branch $branch
+"
+
+git checkout -b ${branch}
```
Comment (mikenairn) [inline scripts/release.sh]: I'd guess it should do a fetch somewhere near the beginning to make sure there isn't already a branch created. Any reason we are letting the user input a branch name? Just seem likely to go wrong the more we ask the person doing it to manually input things.

File: scripts/release.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+if [[ $? > 0 ]]
+then
+exit "command failed" $?
+fi
+
+fi
+
+#check we are upto date and no local changes
+dirty=$(git ls-files -m | wc -l)
```
Comment (mikenairn) [inline scripts/release.sh]: Do we care about untracked files?

File: scripts/release.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+ the local file system is dirty cannot continue
+"
+exit
+fi
+
+echo "
+what release version do you want to create (e.g. release-v1.3.0-rc1)?"
+read  -p "Input: " release
+
+releaseExists=$(git tag | grep release-1.2.0-rc6 | wc -l)
```
Comment (mikenairn) [inline scripts/release.sh]: Hard coded "release-1.2.0-rc6", assume it;s supposed to be release?

Comment (mikenairn) [reply scripts/release.sh]: Again, need to force a fetch somewhere to make sure your local refs are up to date.

File: scripts/release.sh [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+"
+exit
+fi
+
+#update the manifest with the release tag
+sed -i.bak -E "s/^integreatly_version: .*$/integreatly_version: $release/g"  ./evals/inventories/group_vars/all/manifest.yaml && rm ./evals/inventories/group_vars/all/manifest.yaml.bak
+
+#commit the change and push
+git commit -am "release manifest version  update for $release"
+git push origin ${branch}
```
Comment (mikenairn) [inline scripts/release.sh]: Assumes the remote we are pushing to is called "origin", wouldn't always be the case if someone was running this locally.

**Comment by mikenairn** [comment]
I know we discussed using a script like this offline, but having seen it, i have mixed feelings about doing it this way. As it is, it is very dependant on the person running it having everything correct locally, and knowing what format tags/branches should be in. 
Personally i'd much rather the script was more opinionated about what it was going to do when creating a new release and didn't rely on user input so much, in a similar way that the pipeline we currently have just asks for a version tag. I also think even though we have a script in here, we should consider running it in a pipeline anyway to avoid a users local setup becoming an issue. It can still be executed locally if needs be.
Might also be more user friendly if we used python instead of bash, but maybe that's subjective.

**Comment by maleck13** [comment]
@mikenairn Perhaps the right option here is to add flags that will allow it be done non interactively. Then it could hooked in to pipeline easily?

File: scripts/release.sh (hunk 4 above)
Comment (maleck13) [reply scripts/release.sh]: good catch will change

File: scripts/release.sh (hunk 3 above)
Comment (maleck13) [reply scripts/release.sh]: I don't think so?

File: scripts/release.sh (hunk 2 above)
Comment (maleck13) [reply scripts/release.sh]: Yeah a fetch makes sense to ensure we are not creating a branch that already exists

File: scripts/release.sh (hunk 5 above)
Comment (maleck13) [reply scripts/release.sh]: fair point. Can be changed to cover it by a flag

File: scripts/release.sh (hunk 2 above)
Comment (maleck13) [reply scripts/release.sh]: We ask for a gitref currently in the release pipeline

**Comment by david-martin** [comment]
@maleck13 Does this look right?

```
oc get secret manifest -n webapp --template '{{index .data "generated_manifest"}}' | base64 -d
{
    "version": "master",
    "components":[
```
The version is at the top level of the manifest.

Also, I've pushed a fix for the branch creation/check.

**Comment by maleck13** [comment]
@david-martin yes that looks correct

**Comment by david-martin** [comment]
Merging. Lets try this out for 1.3.0

**Pull request merged**